feat: Add Online Page Viewer for Past Paper Preview

amirxhero opened the pull request:
Introduces a new PdfPreviewDialog component for in-app PDF viewing and downloading. Integrates PDF preview functionality into the preview-gallery component, enabling users to preview PDFs directly from the gallery when showDocPreview is enabled. Updates the paper page to pass necessary props for PDF preview support.

amirxhero committed:
feat (components) : Add Online Page Viewer for Past Paper Preview Fixed #535

Introduces a new PdfPreviewDialog component for in-app PDF viewing and downloading. Integrates PDF preview functionality into the preview-gallery component, enabling users to preview PDFs directly from the gallery when showDocPreview is enabled. Updates the paper page to pass necessary props for PDF preview support.

amirxhero committed:
fix (components) : Improve PDF preview handling with object URLs

Replaces direct PDF embedding with conversion to object URLs for better compatibility and error handling. Adds base64 conversion logic, cleans up object URLs on unmount, and improves error messaging for failed previews.

amirxhero committed:
fix (components) : Refactor PDF preview to use vue3-pdfjs

Replaces the previous PDF embed logic in PdfPreviewDialog.vue with vue3-pdfjs for improved rendering and adds zoom/rotate controls. Updates preview-gallery.vue to wrap the PDF dialog in client-only. Adds pdfjs-dist and vue3-pdfjs dependencies.

sanaderi commented on components/details/preview-gallery.vue:
```diff
@@ -78,22 +95,32 @@
       </div>
     </section>
+    <client-only>
+      <common-pdf-preview-dialog
```
Plz change this component to load lazily by adding Lazy at the beginning of the component name.

amirxhero committed:
refactor (components) : Improve PDF preview dialog with virtual scrolling

Refactored PdfPreviewDialog to support virtual scrolling, buffered page rendering, and optimized mobile/desktop layouts. Added page navigation, zoom, and rotation controls for both mobile and desktop. Updated preview-gallery to use lazy-loaded PDF preview dialog component.

amirxhero committed:
refactor (components) : Remove PDF external actions and debug logging

Eliminated buttons for opening and downloading PDFs externally, as well as related state and methods. Also removed various console log statements and debug code to clean up the component.
